pkg/controller/association/controller/ent_es.go

```diff
@@ -25,16 +27,22 @@ const (
 
 func AddEntES(mgr manager.Manager, accessReviewer rbac.AccessReviewer, params operator.Parameters) error {
 	return association.AddAssociationController(mgr, accessReviewer, params, association.AssociationInfo{
-		AssociatedObjTemplate: func() commonv1.Associated { return &entv1beta1.EnterpriseSearch{} },
-		AssociationName:       "ent-es",
-		AssociatedShortName:   "ent",
+		AssociationObjTemplate: func() commonv1.Association { return &entv1beta1.EnterpriseSearch{} },
+		ElasticsearchRef: func(c k8s.Client, association commonv1.Association) (bool, commonv1.ObjectSelector, error) {
+			return true, association.AssociationRef(), nil
+		},
+		ExternalServiceURL:  getElasticsearchExternalURL,
+		AssociatedNamer:     esv1.ESNamer,
+		AssociationName:     "ent-es",
+		AssociatedShortName: "ent",
 		AssociationLabels: func(associated types.NamespacedName) map[string]string {
 			return map[string]string{
 				EntESAssociationLabelName:      associated.Name,
 				EntESAssociationLabelNamespace: associated.Namespace,
 			}
 		},
-		UserSecretSuffix: "ent-user",
+		UserSecretSuffix:         "ent-user",
+		CASecretServiceLabelName: eslabel.ClusterNameLabelName,
```

Comment:
`CASecretLabelName` is probably good enough? (with proper doc in the `AssociationInfo` type)